drools-game-engine-horserace-kjar/src/test/java/org/drools/game/horserace/tests/GameAPITest.java

```diff
@@ -0,0 +1,152 @@
+
+package org.drools.game.horserace.tests;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.drools.game.core.api.GameMessage;
+import org.drools.game.core.api.GameSession;
+import org.drools.game.core.CommandExecutor;
+import org.drools.game.core.BaseGameConfigurationImpl;
+import org.drools.game.core.BasePlayerConfigurationImpl;
+import org.drools.game.core.GameCallbackServiceImpl;
+import org.drools.game.core.GameMessageServiceImpl;
+import org.drools.game.model.impl.base.BasePlayerImpl;
+import org.drools.game.core.GameSessionImpl;
+import org.drools.game.core.api.Command;
+import org.drools.game.core.api.GameCallbackService;
+import org.drools.game.core.api.GameConfiguration;
+import org.drools.game.core.api.GameMessageService;
+import org.drools.game.core.api.PlayerConfiguration;
+import org.drools.game.horserace.cmds.EnterCheckpointCommand;
+import org.drools.game.horserace.cmds.LeaveCheckpointCommand;
+import org.drools.game.horserace.model.Checkpoint;
+
+import org.drools.game.model.api.Player;
+import org.hamcrest.Matchers;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.api.runtime.rule.FactHandle;
+
+/**
+ *
+ * @author salaboy
+ *
+ */
+@RunWith( Arquillian.class )
+public class GameAPITest {
+
+    @Deployment
+    public static JavaArchive createDeployment() {
+
+        JavaArchive jar = ShrinkWrap.create( JavaArchive.class )
+                .addClass( GameSessionImpl.class )
+                .addClass( CommandExecutor.class )
+                .addClass( GameMessageService.class )
+                .addClass( GameMessageServiceImpl.class )
+                .addClass( GameCallbackService.class )
+                .addClass( GameCallbackServiceImpl.class )
+                .addAsManifestResource( EmptyAsset.INSTANCE, "beans.xml" );
+//        System.out.println( jar.toString( true ) );
+        return jar;
+    }
+
+    @Inject
+    private GameSession game;
+
+    /*
+
+     */
+    @Test
+    public void gameAPITest() {
+        Player player = new BasePlayerImpl( "Morton Abenthy Halputz" );
+
+        List initFacts = new ArrayList();
+        
+        Checkpoint startfinish = new Checkpoint("StartFinish", 0, true);
+        initFacts.add( startfinish );
+        Checkpoint checkpointone = new Checkpoint("CheckPointOne", 1, false);
```

Comment:
There should be a constructor that doesn't required the boolean argument, and default to false if it is not provided.